### DISCUSSION TITLE
Overlapping Instances problem

### DISCUSSION THREAD

**Discussion by eldritch-cookie** [haskell-effectful/effectful#269] [Q&A] [answered]
hello i am writing a wrapper for co-log-core and when writing the tests i am stumped [here](https://play.haskell.org/saved/5R3aeXvX) is my somewhat minimised reproducer, i understand this is because of ambiguous type variables but i tried some variantions of TypeApplications which didn't work.

**Comment by eldritch-cookie** [comment, marked as answer]
the plugin solves this issue

**Comment by arybczak** [comment]
For the record, if you want to publish this as a library, it'd be better to make the effect dynamic, more flexibility for users.

**Reply by eldritch-cookie** [reply]
how exactly does a dynamic Log Effect provides more flexibility? 
the runner takes a LogAction in the Eff Monad and i can't think of anything you could do with dynamic that you can't by just modifying the LogAction. I guess you could implement new handlers but i can't come up with any handlers which are meaningful besides the one that takes a LogAction.

**Reply by arybczak** [reply]
Fair enough, I don't know the API of colog and thought you made it a minimal reproducer to show off the problem. If that's all it happens they yeah, dynamic dispatch wouldn't add much.
